### ISSUE TITLE
why skip_diff_train before fast_epochs

### ISSUE THREAD

**Issue by wblgers** [PlayVoice/Grad-SVC#12]
Hi,

Can you explain why skip diffusion train before the configured fast_epochs?
And how many epochs does diffusion train need?

Thanks!

**Comment by MaxMax2016** [comment]
因为刚开始训练的时候，encoder还学的不好。encoder和diffusion都在拟合mel谱图，没有diffusion encoder也可以独立工作，整合系统甚至可以把diffusion完全删除也是可以的。等encoder学好了，在加入diffusion训练，可以节约训练时间。diffusion训两百个epoch可以看到效果。

**Comment by wblgers** [comment]
> 因为刚开始训练的时候，encoder还学的不好。encoder和diffusion都在拟合mel谱图，没有diffusion encoder也可以独立工作，整合系统甚至可以把diffusion完全删除也是可以的。等encoder学好了，在加入diffusion训练，可以节约训练时间。diffusion训两百个epoch可以看到效果。

谢谢你的解答，另外有一个问题。为什么SpeakerClassifier中的说话人个数是写死的？line 292 of encoder.py
`
self.speaker = SpeakerClassifier(
            n_channels,
            256,  # n_spks: 256
        )
`

**Comment by MaxMax2016** [comment]
256是声纹向量的维度

**Comment by wblgers** [comment]
> 256是声纹向量的维度

好的，谢谢解答！

**Comment by wblgers** [comment]
@MaxMax2016 请教一个问题，你是多卡还是单卡训练的？
我自己写了一个多卡的代码，训练速度上来了，但是mel_loss收敛的并没有单卡好。
单卡训练到250epoch大概是：
Epoch 250: | spk loss = 0.337 | mel loss = 0.330 | prior loss = 1.139 | diffusion loss = 0.097
多卡训练到250epoch大概是：
Epoch 250: | spk loss = 0.335 | mel loss = 0.335 | prior loss = 1.135 | diffusion loss = 0.099

**Comment by MaxMax2016** [comment]
单卡，改用多卡要调学习率的吧

**Comment by wblgers** [comment]
> 单卡，改用多卡要调学习率的吧
调了学习率之后，直接跑飞了，melloss一直0.72不下降
